openedx/core/lib/edx_api_utils.py

```diff
@@ -56,11 +56,16 @@ def get_edx_api_data(api_config, resource, api, resource_id=None, querystring=No
 
         cached = cache.get(cache_key)
         if cached:
-            cached_response = zunpickle(cached)
-            if fields:
-                cached_response = get_fields(fields, cached_response)
-
-            return cached_response
+            try:
+                cached_response = zunpickle(cached)
+                if fields:
+                    cached_response = get_fields(fields, cached_response)
+
+                return cached_response
+            except Exception:
```

Comment:
I understand why you want the try to be this broad, but for the specific case we're dealing with this week, i would put the try/except around just the zunpickle.

Reply:
I pulled out everything except for the zunpickle into an else clause.